nx/lib/nx/random.ex

```diff
@@ -551,6 +551,181 @@ defmodule Nx.Random do
     |> stop_grad()
   end
 
+  @doc """
+  Returns a sample from a multivariate normal distribution with given `mean` and `covariance` (matrix).
+
+  ## Options
+
+    * `:type` - a float type for the returned tensor
+
+    * `:shape` - shape of the returned tensor
+
+    * `:names` - the names of the returned tensor
+
+  ## Examples
+
+      iex> key = Nx.Random.key(12)
+      iex> {multivariate_normal, _new_key} = Nx.Random.multivariate_normal(key, Nx.tensor([0]), Nx.tensor([[1]]))
+      iex> multivariate_normal
+      #Nx.Tensor<
+        f32[1]
+        [0.735927939414978]
+      >
+
+      iex> key = Nx.Random.key(12)
+      iex> {multivariate_normal, _new_key} = Nx.Random.multivariate_normal(key, Nx.tensor([0, 0]), Nx.tensor([[1, 0], [0, 1]]))
+      iex> multivariate_normal
+      #Nx.Tensor<
+        f32[2]
+        [-1.3425945043563843, -0.40812060236930847]
+      >
+
+      iex> key = Nx.Random.key(12)
+      iex> {multivariate_normal, _new_key} = Nx.Random.multivariate_normal(key, Nx.tensor([0]), Nx.tensor([[1]]), shape: {3, 2}, type: :f16)
+      iex> multivariate_normal
+      #Nx.Tensor<
+        f16[3][2][1]
+        [
+          [
+            [0.326904296875],
+            [0.2176513671875]
+          ],
+          [
+            [0.316650390625],
+            [0.1109619140625]
+          ],
+          [
+            [0.53955078125],
+            [-0.8857421875]
+          ]
+        ]
+      >
+
+      iex> key = Nx.Random.key(12)
+      iex> {multivariate_normal, _new_key} = Nx.Random.multivariate_normal(key, Nx.tensor([0, 0]), Nx.tensor([[1, 0], [0, 1]]), shape: {3, 2})
+      iex> multivariate_normal
+      #Nx.Tensor<
+        f32[3][2][2]
```

Comment:
I'm a bit confused about these output shapes. I guess they are related to the input tensor shapes, but given the rest of the behavior in Nx.Random, I'd expect the :shape option to fully control the output shape

Reply:
The last dimension is equal to the dimension of the distribution (in this case 2, as both mean and covariance have dimensions 2). If shape option is not provided (i.e. `nil`), then the output will be of shape `{dim}`.
If shape option is `{3, 2}`, then the output will be of shape `{3, 2, 2}`.
This is the same as in NumPy and JAX.
https://github.com/numpy/numpy/blob/104addf377577d12ea34547303822369fcb6e44e/numpy/random/_generator.pyx#L3777
https://jax.readthedocs.io/en/latest/_autosummary/jax.random.multivariate_normal.html#jax.random.multivariate_normal

Reply:
The docs can use some improvement to clarify this. Something along the lines of the first 3 phrases of this answer should suffice :)

Reply:
Agreed!

Reply:
And done!